packages/@lwc/template-compiler/src/parser/simple-expression-parser/index.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { invariant, ParserDiagnostics } from '@lwc/errors';
+import { ASTExpression, ASTIdentifier } from './types';
+import { validateParsedExpression } from './validation';
+import { IRNode } from '../../shared/types';
+
+const WHITESPACE = new Set([' ', '\t', '\v', '\f', '\r', '\n']);
+
+interface ExpressionParser {
+    position: number;
+    peek(): string;
+    match(expected: string): boolean;
+    eat(expected?: string): string;
+}
+
+function isValidIdentifierStart(char: string): boolean {
```

Comment:
Also, this is being used to validate characters at any position in the identifier, not just the start